lib/spack/spack/test/url_fetch.py

```diff
@@ -354,27 +354,12 @@ def test_url_missing_curl(mutable_config, missing_curl, monkeypatch):
         web_util.url_exists("https://example.com/")
 
 
-def test_url_fetch_text_urllib_bad_returncode(mutable_config, monkeypatch):
```

Comment:
this test is based on a wrong assumption that 404 http errors do not raise, they do. and raising errors is the next test, making this redundant.